### ISSUE TITLE
The getPendingMigrations creates migrations table

### ISSUE THREAD

**Issue by pkuczynski** [typeorm/typeorm#10858]
### Issue description

The getPendingMigrations creates migrations table

### Expected Behavior

When I call `getPendingMigrations` I expect it would check if migrations table exists and compare it's content with my migrations folder to tell me, which migrations are not executed yet. If the table does not exist, it should return all of the migrations as pending.

### Actual Behavior

`getPendingMigrations` does create the `migrations` table instead of checking if it exists before reading it.

### Steps to reproduce

Run `getPendingMigrations` on the empty database:

```
const migrationExecutor = new MigrationExecutor(dataSource);
await migrationExecutor.getPendingMigrations();
```

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |          |
| Node.js version     | 20  |
| Typescript version  | 5.3.3  |
| TypeORM version     | 0.3.19  |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

Yes, I have the time, and I know how to start.

**Comment by pkuczynski** [comment]
Yes, this issue is still relevant. It was stale since nobody from the team commented on it. If you confirm, I could come up with PR.

**Comment by pkuczynski** [comment]
@gioboa should I open a new issue or this one will be reviewed again?

**Comment by gioboa** [comment]
> Yes, this issue is still relevant. It was stale since nobody from the team commented on it. If you confirm, I could come up with PR.

Thanks for the feedback. Feel free to drop a PR for this, appreciated. 🙏

**Comment by pkuczynski** [comment]
Here you go @gioboa https://github.com/typeorm/typeorm/pull/11672. On this occasion I also created #11673, which I am happy to work on a fix.
